### PR TITLE
bm43a stevexyz keymap

### PR DESCRIPTION
**Pull request by stevexyz**
A full featured keymap for this little nice keybard as an alternative to the very basic standard one.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

None. New keymap. 

## Checklist

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/bm43a/keymaps/stevexyz/config.h [hunk 1]
```diff
@@ -0,0 +1,19 @@
+/* Copyright 2019 mechmerlin
```
Comment (zvecr) [inline keyboards/bm43a/keymaps/stevexyz/config.h:1]: This file can be removed

File: keyboards/bm43a/keymaps/stevexyz/rules.mk [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+RGBLIGHT_ENABLE = yes       # Enable keyboard RGB underglow
+
+SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
+NKRO_ENABLE = no            # USB Nkey Rollover
+MIDI_ENABLE = no            # MIDI support
+UNICODE_ENABLE = no         # Unicode
+BLUETOOTH_ENABLE = no       # Enable Bluetooth with the Adafruit EZ-Key HID
+AUDIO_ENABLE = no           # Audio output on port C6
+FAUXCLICKY_ENABLE = no      # Use buzzer to emulate clicky switches
+HD44780_ENABLE = no         # Enable support for HD44780 based LCDs
```
Comment (zvecr) [inline keyboards/bm43a/keymaps/stevexyz/rules.mk:20]: This file should only contain overrides,
```suggestion
BOOTMAGIC_ENABLE = no       # Virtual DIP switch configuration
COMMAND_ENABLE = no         # Commands for debug and configuration
CONSOLE_ENABLE = no         # Console for debug

MOUSEKEY_ENABLE = yes       # Mouse keys
EXTRAKEY_ENABLE = yes       # Audio control and System control
BACKLIGHT_ENABLE = yes      # Enable keyboard backlight functionality
RGBLIGHT_ENABLE = yes       # Enable keyboard RGB underglow
```

File: keyboards/bm43a/keymaps/stevexyz/config.h (hunk 1 above)
Comment (stevexyz) [reply keyboards/bm43a/keymaps/stevexyz/config.h]: done! :)

File: keyboards/bm43a/keymaps/stevexyz/keymap.c [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+    DF(0), KC_SLEP, KC_BRID, KC_BRIU, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, RESET,
+    XXXXXXX, BL_TOGG, BL_DEC, BL_INC, BL_STEP, BL_BRTG, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX,
+    XXXXXXX, RGB_TOG, RGB_RMOD, RGB_MOD, RGB_VAD, RGB_VAI, RGB_HUD, RGB_HUI, RGB_SAD, RGB_SAI, XXXXXXX,
+    XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX ),
+};
+
+// set num lock alwways on
+void matrix_init_user(void) {
+  if (!host_keyboard_led_state().num_lock)
+    tap_code(KC_LOCKING_NUM); //KC_NUMLOCK
```
Comment (fauxpark) [inline keyboards/bm43a/keymaps/stevexyz/keymap.c:44]: Are you sure this works? For me on Windows the locking Num Lock key acts more like a momentary key (which makes sense as it's meant to be used with a physically locking switch).

Comment (stevexyz) [reply keyboards/bm43a/keymaps/stevexyz/keymap.c:44]: You're right well spot! And in fact I tried both thi KC_LOCKING_NUM and KC_NUMLOCK without success and to make it working I substituted the KC_Px number codes with standard numbers.
If there is a way to make it work I can put back the KC_Pnumbers, else this is dead code and can be removed.

Comment (stevexyz) [reply keyboards/bm43a/keymaps/stevexyz/keymap.c:44]: PS: I'm not using Windows on my computers! LOL :)

Comment (fauxpark) [reply keyboards/bm43a/keymaps/stevexyz/keymap.c:44]: I think `matrix_init_user()` is a bit too early to be doing this. Try `keyboard_post_init_user()`, and if that doesn't work, just do it in `led_update_user()` (I believe the host sends an initial lock LED state at which point it's probably safe to send keypresses).

Comment (stevexyz) [reply keyboards/bm43a/keymaps/stevexyz/keymap.c]: Triead in various ways but didn't work. I've removed the code.

**Pull request merged**